Tests/ClientTests.swift

```diff
@@ -302,6 +302,28 @@ class ClientTests: OnlineTestCase {
         
         self.waitForExpectations(timeout: expectationTimeout, handler: nil)
     }
+  
+  func testLongAPIKey() {
+    let validAPIKey = self.client.apiKey
+    defer {
+      // Restore the valid API key (otherwise tear down will fail).
+      self.client.headers["X-Algolia-API-Key"] = validAPIKey
```

Comment:
You can access the internal `AbstractClient.xAlgoliaAPIKey` in your tests, so you can use it here and down as well